### PR TITLE
fix oci:// chart

### LINKED ISSUE #473: OCI charts do not work with helmfile fetch
### Operating system

Windows 11/10

### Helmfile Version

0.147.0

### Helm Version

3.10.1

### Bug description

helmfile fetch does not download and untar OCI charts if the chart is not referenced with `repositories`

### Example helmfile.yaml

```
# KO for fetch but other commands work
releases:
- name: nginx
  chart: oci://ghcr.io/itscaro/charts/nginx
  version: 13.2.12
  values: []
```

```
# OK
repositories:
- name: oci-name
  url: ghcr.io/itscaro/charts
  oci: true
releases:
- name: nginx
  chart: oci-name/nginx
  version: 13.2.12
  values: []
```

### Error message you've seen (if any)

```
helmfile fetch --debug
processing file "helmfile.yaml" in directory "."
changing working directory to "C:\Users\qtran\workspace\github\helmfile\debug\oci"
first-pass rendering starting for "helmfile.yaml.part.0": inherited=&{default map[] map[]}, overrode=<nil>
first-pass uses: &{default map[] map[]}
first-pass rendering output of "helmfile.yaml.part.0":
 0: releases:
 1: - name: nginx
 2:   chart: oci://ghcr.io/itscaro/charts/nginx
 3:   version: 13.2.12
 4:   values: []
 5:

first-pass produced: &{default map[] map[]}
first-pass rendering result of "helmfile.yaml.part.0": {default map[] map[]}
vals:
map[]
defaultVals:[]
second-pass rendering result of "helmfile.yaml.part.0":
 0: releases:
 1: - name: nginx
 2:   chart: oci://ghcr.io/itscaro/charts/nginx
 3:   version: 13.2.12
 4:   values: []
 5:

merged environment: &{default map[] map[]}
helm.exe:SlCPF> v3.10.1+g9f88ccb
changing working directory back to "C:\Users\qtran\workspace\github\helmfile\debug\oci"
```

```
helmfile template
Templating release=nginx, chart=oci://ghcr.io/itscaro/charts/nginx
Pulled: ghcr.io/itscaro/charts/nginx:13.2.12
Digest: sha256:c657835bdf40f5d23b812ce9092220b4016694a41dcbc0e3415f2dda57cf067c
```

### Steps to reproduce

N/A

### Working Helmfile Version

N/A

### Relevant discussion

_No response_

### PR DESCRIPTION
**Pull request by itscaro**
fixes #473

Fix for chart which is referenced with oci:// instead of repository alias

### CODE REVIEW

**Comment by yxxhero** [comment]
@itscaro could you add some tests for that issue?

**Comment by itscaro** [comment]
> @itscaro could you add some tests for that issue?

It's better to have e2e test,  I don't know how this works could you help me for the test? I will add some unit test for the change as well. Thanks

**Comment by yxxhero** [comment]
@itscaro yes. I will help you to add e2e test.

**Comment by yxxhero** [comment]
@itscaro unit test has an issue.

**Comment by itscaro** [comment]
@yxxhero fixed

**Comment by yxxhero** [comment]
@itscaro OK. I will add an e2e test for this PR. Then merge.

**Comment by yxxhero** [comment]
@itscaro done

**Comment by yxxhero** [comment]
@itscaro ko in fetch command？

**Review by yxxhero — APPROVED** [review]
LGTM

File: test/e2e/template/helmfile/testdata/snapshot/issue_473_oci_chart_url_fetch/output.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Pulling localhost:5000/myrepo/raw:0.1.0
+Pulled: localhost:5000/myrepo/raw:0.1.0
+Digest: $DIGEST
```
[the remark this answers is not in the document]
Comment (yxxhero) [reply test/e2e/template/helmfile/testdata/snapshot/issue_473_oci_chart_url_fetch/output.yaml:3]: @itscaro yes. I added https://github.com/helm/helm/pull/11480 into 3.10.2

**Pull request merged**